### PR TITLE
chore: add fido2-lib

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

**Comment by lewisdaly** [comment]
@kleyow  I managed to get it working! There was some strange encoding stuff going on between `str2ab`, and what the front end client was doing.

It's now working, we just needed to base64encode the challenge in the `attestationExpectations` object in the end. I also fixed the front end client to encode the challenge to a `utf-8` ArrayBuffer (per the webauthn.guide example), instead of `utf-16`

**Comment by kleyow** [comment]
Awesome. This saves us a ton of work. I can now write the registration/database logic/step into the RegisterConsent flow.

**Review by elnyry-sam-k — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -38,8 +38,8 @@
     "strictPropertyInitialization": true,
     "noImplicitThis": true,
     "alwaysStrict": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
+    "noUnusedLocals": false,
```
Comment (lewisdaly) [inline tsconfig.json:41]: Can we put this back please? I was being lazy

**Review by lewisdaly — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 1 above)
Comment (kleyow) [reply tsconfig.json]: done

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**